bazel/carbon_rules/defs.bzl

```diff
@@ -0,0 +1,40 @@
+# Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+# Exceptions. See /LICENSE for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""Provides rules for building Carbon files using the toolchain."""
+
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+
+def carbon_binary(name, srcs):
+    """Compiles a Carbon binary.
+
+    Args:
+      name: The name of the build target.
+      srcs: List of Carbon source files to compile.
+    """
+    carbon_deps = ["//core:prelude", "//core:prelude_deps"] + srcs
+    for src in srcs:
+        # Build each source file. For now, we pass all sources to each compile
+        # because we don't have visibility into dependencies and have no way to
+        # specify multiple output files. Object code for each input is written
+        # into the output file in turn, so the final carbon source file
+        # specified ends up determining the contents of the object file.
+        #
+        # We also pass in the prelude files, but not prelude.carbon because the
+        # driver adds that itself. For now, we assume that the prelude doesn't
+        # produce any necessary object code, and don't include the .o files for
+        # //core/prelude... in the final linked binary.
```

Comment:
I'm concerned what you're doing here is going to be brittle. Your changes to prelude.carbon affect file_test, which needs to also load any dependencies. Right now that's handled through the driver. By creating a divergence here, I expect file_test to break.

Had you considered instead making dependencies of the main prelude file load the same way that the main prelude file is loaded?

Reply:
Done.